components/org.wso2.carbon.config.configuration/src/main/java/org/wso2/carbon/config/configuration/internal/ConfigProviderDataHolder.java

```diff
@@ -44,12 +42,4 @@ private ConfigProviderDataHolder() {
     public void setSecureVault(SecureVault secureVault) {
         this.optionalSecureVault = Optional.ofNullable(secureVault);
     }
-
-    public Optional<BundleContext> getBundleContext() {
-        return bundleContext;
-    }
-
-    public void setBundleContext(BundleContext bundleContext) {
-        this.bundleContext = Optional.ofNullable(bundleContext);
-    }
```

Comment:
We need to keep the reference of the BundleContext. we need bundleContext to differentiate osgi and non-osgi modes